swap/peer.go

```diff
@@ -74,15 +74,10 @@ func (sp *Peer) handleMsg(ctx context.Context, msg interface{}) error {
 // a cheque from a creditor
 // TODO: validate the contract address in the cheque to match the address given at handshake
 // TODO: this should not be blocking
-func (sp *Peer) handleEmitChequeMsg(ctx context.Context, msg interface{}) error {
+func (sp *Peer) handleEmitChequeMsg(ctx context.Context, msg *EmitChequeMsg) error {
 	log.Info("received emit cheque message")
 
-	chequeMsg, ok := msg.(*EmitChequeMsg)
-	if !ok {
-		return fmt.Errorf("Invalid message type, %v", msg)
-	}
-
-	cheque := chequeMsg.Cheque
+	cheque := msg.Cheque
 	if cheque.Contract != sp.contractAddress {
```

Comment:
Wouldn't it be wise to rename cheque.Contract to cheque.contractAddress or cheque.address?
Not a big issue for me, but I noticed this already in lake Balaton and had to look twice here as well what exactly cheque.Contract is.

Reply:
I agree it is confusing.

Nevertheless there is apparently a `go` guideline which suggests to not add the type into a variable name. `Contract` is of type `common.Address`.

Nevertheless we are not very consistent with this notion either, because `Owner.address` is also a `common.Address` - I would rather like to change *that* name. Any suggestion?